pkg/utils/deployer.go

```diff
@@ -479,8 +479,7 @@ func ApplyResourceWithRetry(ctx context.Context, dynamicClient dynamic.Interface
 		} else {
 			lastError = nil
 		}
-
-		if ResourceNeedResync(curObj, resource) {
+		if ResourceNeedResync(resource, curObj) {
```

Comment:
"jsonpatch.CreatePatch(a, b []byte) 'a' is original, 'b' is the modified document. so ResourceNeedResync(resource, curObj) is wrong and will result in unnecessary update event. Just exchange the position of the two parameter."

for our code, `curObj` is the original resource, `resource` is the modified resource

Reply:
Yeah, i get the point, surely need more consideration.